### ISSUE TITLE
mu commander mac os - queue

### ISSUE THREAD

**Issue by srdm1977** [mucommander/mucommander#700]
Hi there, I´m using this great app for my project-backups on local drives, dropbox and ftp. Updating a backup takes pretty much time, as mu commander checks file by file if it fits to the condition: replace different file size. Would it be possible to see a queue to have an info, how many files will be copied? And maybe a calculation, how long this will take?

**Comment by ahadas** [comment]
@srdm1977 displaying a reliable progress based on the size of the data being transferred or the number of files is tricky - it means we'll need to analyze nested folders first in order to know that and then to go over the same list to copy the data. different heuristics for displaying progress that are not "reliable" could end up in this situation that I remember when I used Windows long time ago - that it can go, for example, from 0 to 80% in 5 min and then from 80% to 100% in 30 min. I think that showing such an unreliable progress doesn't worth the effort and having a reliable one could have a significant performance penalty. do you have a suggestion for something in between that could work?

**Comment by Andreas0602** [comment]
> it means we'll need to analyze nested folders first in order to know that and then to go over the same list to copy the data

My vote goes to that solution. By the way, I've opened already another issue about the progress bar six months ago: #565

**Comment by ahadas** [comment]
@Andreas0602 yeah, I've noticed #565, thanks for taking the time to file it. it's related to this one indeed because you pointed out a specific case there in which the current logic fails to estimate the progress correctly. however, in the trade off between providing a good progress and saving time (and with some protocol may also be cost), I'd go with the latter. maybe we can implement it differently for local files, for which is should typically be fairly quick, though
